python/ray/serve/deployment.py

```diff
@@ -67,6 +95,32 @@ def __getattr__(self, name: str) -> Any:
 
 @PublicAPI
 class Deployment:
+    """Defines group of Ray actors to be deployed in an `Application`.
+
+    Deployments are used to define a Serve application by binding them together with
+    arguments (and possibly other deployments for composition).
+
+    The resulting `Application` can be deployed using `serve.run` or via a Serve config
+    file.
```

Comment:
The summary line "Defines group of Ray actors to be deployed in an `Application`." feels a bit confusing to me. 

Also, the description is talking a lot about applications - would it be better to describe more concretely what a deployment is, e.g. "created by using the @serve.deployment decorator" or "can either be a function or class"? Maybe mention at the very end that the deployments can be composed together to form an application?